be/src/util/owned_slice.h

```diff
@@ -0,0 +1,60 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "gutil/macros.h"
+
+namespace doris {
+
+struct Slice;
+
+class OwnedSlice {
+
+public:
+    Slice slice = nullptr;
```

Comment:
Slice is not a pointer, better not be assigned from nullptr.
And better to make slice private

Reply:
yep,I meet a core dump,this is a fatal error

Reply:
>And better to make slice private 

to prevent others to change the slice?

Reply:
Yes, you have already prevented copy and assign, it is reasonable to prevent changing it

Reply:
maybe set OwnedSlice's slice to const is better?

Reply:
when doing move,we just change slice.data, while not slice

Reply:
slice data is changed, slice is changed too

Reply:
👌